sdk/keyvault/keyvault/inc/az_keyvault.h

```diff
@@ -5,43 +5,103 @@
 #define AZ_KEYVAULT_H
 
 #include <az_contract.h>
+#include <az_credential.h>
+#include <az_http_pipeline.h>
+#include <az_http_request_builder.h>
 #include <az_result.h>
 #include <az_span.h>
+#include <az_str.h>
 
 #include <stdlib.h>
 
 #include <_az_cfg_prefix.h>
 
-typedef struct {
-  az_span uri;
-  // TODO: Azure Credentials
-} az_keyvault_keys_client;
+extern az_span const AZ_KEY_VAULT_NONE_TYPE;
+extern az_span const AZ_KEY_VAULT_KEY_TYPE;
+extern az_span const AZ_KEY_VAULT_SECRET_TYPE;
+extern az_span const AZ_KEY_VAULT_CERTIFICATE_TYPE;
+extern az_span const AZ_KEY_VAULT_URL_PATH_SEPARATOR;
+typedef enum {
+  AZ_KEY_VAULT_NONE = 0,
+  AZ_KEY_VAULT_KEY = 1,
+  AZ_KEY_VAULT_SECRET = 2,
+  AZ_KEY_VAULT_CERTIFICATE = 3,
+} az_key_vault_key_type;
 
 typedef struct {
-  // API version
-  // Pipeline configuration options
   az_span version;
 } az_keyvault_keys_client_options;
 
+typedef struct {
+  az_http_policy pipeline[8];
```

Comment:
policies

Reply:
I agree with Rick's name suggestion here and the next one